pilot/pkg/model/push_context.go

```diff
@@ -226,8 +226,30 @@ type PushRequest struct {
 	// Start represents the time a push was started. This represents the time of adding to the PushQueue.
 	// Note that this does not include time spent debouncing.
 	Start time.Time
+
+	// Reason represents the reason for requesting a push. This should only be a fixed set of values,
+	// to avoid unbounded cardinality in metrics. If this is not set, it may be automatically filled in later.
+	// There may be multiple reasons, only if there are multiple requests merge together
+	Reason []TriggerReason
 }
 
+type TriggerReason string
+
+var (
```

Comment:
nit: "Update" seems redundant in the name. "Endpoint" ,"Service"... "Unknown" like that may be simpler.

Reply:
I think its not because its the in the model package. `model.Unknown` is not descriptive enough

Reply:
Another idea, Prefixing these constants with `TriggerReasonUnknown` or `TriggerReason_Unknown`. Seen some usages like that in K8S code base and in the golang pb files generated like `CaptureMode_Default` for capture mode - some thing to consider